pkg/engine/params_k8s.go

```diff
@@ -154,6 +153,17 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				}
 			}
 		}
+		// The cluster is AKS-hosted cluster and it uses managed identity
+		if kubernetesConfig.UseManagedIdentity && properties.IsHostedMasterProfile() {
+			servicePrincipalProfile := properties.ServicePrincipalProfile
+			if servicePrincipalProfile == nil || servicePrincipalProfile.ClientID == "" || servicePrincipalProfile.Secret == "" {
+				addValue(parametersMap, "servicePrincipalClientId", "msi")
```

Comment:
Isn't it safer to error out in this case? Burying AKS default behavior in aks-engine will cause confusion down the road

Reply:
The SPN will soon disappear in AKS when managed identity is enabled. That is `servicePrincipalProfile == nil || servicePrincipalProfile.ClientID == "" || servicePrincipalProfile.Secret == ""`  will become true soon. We can't produce an error in this case otherwise all clusters enabled managed identity will not work.

Reply:
I got your point. It makes sense to add the default value in aks-rp instead of aks-engine, I will add this in aks-rp.